### PR TITLE
XCode project Apple Silicon compatibility

### PR DESCRIPTION
**Pull request by axelmolch**
This PR only aims to add Apple Silicon compatibility to XCode project (as per https://fitbit.github.io/golden-gate/platforms/ios.html)

Please see Brewfile note on `mynewt-newt` packate
Please check `inv native.test` in your system, this PR does not aim to fix or update C unittests.
PR changes XCode project to use XCFrameworks, makes corresponding project structure changes, removes some Carthage build scripts

### CODE REVIEW

File: Brewfile [hunk 1]
```diff
@@ -3,6 +3,7 @@ tap "homebrew/homebrew-bundle"
 tap "homebrew/cask-drivers"
 tap "JuulLabs-OSS/mynewt"
 
+# Note: Please comment out next line for Apple Silicon setup. Proper solution penging.
```
Comment (davidahouse) [inline Brewfile:6]: Nit: `penging` should be `pending`

**Review by davidahouse — COMMENTED** [review]
Looks good to me other than 1 nit. We do need to get a 👍 from someone on an Intel Mac just to verity this hasn't broken the build for them.

File: Brewfile (hunk 1 above)
Comment (axelmolch) [reply Brewfile]: fixed, thanks

**Comment by axelmolch** [comment]
checked iOS target on iOS device

**Review by mikehuovila — APPROVED** [review]
The PR builds the iOS host app from Intel Mac for simulated target and real target.

**Review by davidahouse — APPROVED** [review]
👍

File: external/rxbit/Tools/carthage-copy-and-verify [hunk 2]
```diff
@@ -1,19 +0,0 @@
-#!/bin/sh
```
Comment (emixb) [inline external/rxbit/Tools/carthage-copy-and-verify:1]: Does this mean we'll no longer support developer mode (aka building dependencies from sources)?

File: tasks/apple/carthage.py [hunk 3]
```diff
@@ -94,37 +94,19 @@ def _rome_upload_cmd(ctx, platform):
             --cache-prefix {}".format(platform, _rome_prefix(ctx))
 
 def _carthage_bootstrap_cmd(ctx):
-    # Applying Carthage build workaround to exclude Apple Silicon binaries.
-    # See https://github.com/Carthage/Carthage/issues/3019 for more details
-    fp = NamedTemporaryFile(delete=False, prefix='static.xcconfig.')
 
     xcode_version = find_xcode_version(ctx)
```
Comment (emixb) [inline tasks/apple/carthage.py:98]: These variable seem no longer used. Can you also clean up find_xcode_version, if possible?

Comment (axelmolch) [reply tasks/apple/carthage.py:98]: that means we don't want to print Xcode version too (2 lines below) correct?

File: external/rxbit/Tools/carthage-copy-and-verify (hunk 2 above)
Comment (axelmolch) [reply external/rxbit/Tools/carthage-copy-and-verify:1]: I will ask more questions about it in direct chat, thanks

**Review by barbibulle — APPROVED** [review]

File: tasks/apple/carthage.py (hunk 3 above)
Comment (emixb) [reply tasks/apple/carthage.py:98]: Yes, I don't think those logs are relevant anymore (as we no longer do carthage workaround)

**Review by emixb — APPROVED** [review]

**Comment by axelmolch** [comment]
Manual merge bypassing CI failure "CI / GG-Build macos-latest (pull_request)" as discussed

**Pull request merged**